workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityWorkerMenuGenerator.tt

```diff
@@ -62,21 +62,62 @@ var workerType = workerTypeList[i];
             WorkerBuilder.Clean();
             Debug.Log("Clean completed");
         }
-        
-        private static void MenuBuild(BuildEnvironment environment, params string[] platforms)
+
+        private static void MenuBuild(BuildEnvironment environment, params string[] workerTypes)
         {
             // Delaying build by a frame to ensure the editor has re-rendered the UI to avoid odd glitches.
             EditorApplication.delayCall += () =>
             {
+                foreach (var workerType in workerTypes)
+                {
+                    WorkerBuilder.AssertWorkerCanBuildForEnvironment(workerType, environment);
+                }
+
                 LocalLaunch.BuildConfig();
 
-                foreach (var platform in platforms)
+                foreach (var workerType in workerTypes)
                 {
-                    WorkerBuilder.BuildWorkerForEnvironment(platform, environment);
+                    WorkerBuilder.BuildWorkerForEnvironment(workerType, environment);
                 }
 
                 Debug.LogFormat("Completed build for {0} target", environment);
             };
         }
+
+        [MenuItem(EditorConfig.ParentMenu + "/Validate Build Support", false, EditorConfig.MenuOffset + <#= workerTypeList.Count + 1 #>)]
```

Comment:
Is this something we would recommend for people to run to help diagnose issues?

Reply:
I think so. We could extend it further to turn into a "full diagnose" if we can find other issues to resolve.